packages/alfa-rules/src/sia-r113/rule.ts

```diff
@@ -50,96 +58,133 @@ export default Rule.Atomic.of<Page, Element>({
+ * 1) the circle intersects with the bounding box of any other target, or
+ * 2) the distance between the center of the bounding box of the target
+ *    and the center of the bounding box of any other **undersized** target is less than 24.
  */
 function hasSufficientSpacing(
   document: Document,
   device: Device,
 ): Predicate<Element> {
   return (target) => {
-    // The target might have been cached while computing the spacing for another target
-    const cachedTarget = spacingCache
-      .get(document, Cache.empty)
-      .get(device, Cache.empty)
-      .get(target);
-
-    if (cachedTarget.isSome()) {
-      return cachedTarget.get();
-    }
-
     // Existence of a bounding box is guaranteed by applicability
-    const box = target.getBoundingBox(device).getUnsafe();
+    const targetRect = target.getBoundingBox(device).getUnsafe();
 
-    for (const otherTarget of targetsOfPointerEvents(document, device)) {
-      if (target === otherTarget) {
+    const undersizedTargets = undersizedCache
+      .get(document, Cache.empty)
+      .get(device, () =>
+        targetsOfPointerEvents(document, device).reject(
+          hasSufficientSize(24, device),
+        ),
+      );
+
+    for (const candidate of targetsOfPointerEvents(document, device)) {
+      if (target === candidate) {
         continue;
       }
 
       // Existence of a bounding box is guaranteed by applicability
-      const other = otherTarget.getBoundingBox(device).getUnsafe();
+      const candidateRect = candidate.getBoundingBox(device).getUnsafe();
 
-      // If the box of the target doesn't intersect with the circumscribed square of the other target, we know they are far enough apart
       if (
-        !box.intersects(
-          Rectangle.of(other.center.x - 12, other.center.y - 12, 24, 24),
+        circleIntersectsRect(
+          targetRect.center.x,
+          targetRect.center.y,
+          12,
+          candidateRect,
         )
       ) {
-        continue;
+        return false;
       }
 
-      // TODO: Check if the 24px diameter circle of the target intersect with the bounding box of the other target
-
-      // If the other target is undersized, the 24px diameter circle of the target must not intersect with the 24px diameter circle of the other target
       if (
-        (other.width < 24 || other.height < 24) &&
-        (box.center.x - other.center.x) ** 2 +
-          (box.center.y - other.center.y) ** 2 <=
-          24 ** 2
+        undersizedTargets.includes(candidate) &&
+        distanceSquared(targetRect, candidateRect) < 24 ** 2
       ) {
-        // If the other is undersized and too close to this we already know it will also fail the rule, so we might as well cache it
-        spacingCache
-          .get(document, Cache.empty)
-          .get(device, Cache.empty)
-          .set(otherTarget, false);
-
         return false;
       }
     }
 
     return true;
   };
 }
+
+function circleIntersectsRect(
```

Comment:
**Suggestion:** I feel this function is complex enough to have its own tests.

Also, if it's possible to have an ASCII art explanation of the geometry, that would be awesome…